internal/signerverifier/signerverifier.go

```diff
@@ -0,0 +1,43 @@
+package signerverifier
+
+import (
+	"encoding/json"
+
+	"github.com/adityasaky/gittuf/internal/signerverifier/common"
+	"github.com/adityasaky/gittuf/internal/signerverifier/ed25519"
+	"github.com/adityasaky/gittuf/internal/tuf"
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+)
+
+func NewSignerVerifierFromTUFKey(key *tuf.Key) (dsse.SignerVerifier, error) {
+	switch key.KeyType {
+	case ed25519.Ed25519KeyType:
+		return ed25519.NewSignerVerifierFromTUFKey(key)
+	}
+	return nil, common.ErrUnknownKeyType
+}
+
+func NewSignerVerifierFromSecureSystemsLibFormat(keyContents []byte) (dsse.SignerVerifier, error) {
+	type keyval struct {
+		Public  string `json:"public"`
+		Private string `json:"private"`
+	}
+	k := struct {
+		KeyType             string   `json:"keytype"`
+		KeyID               string   `json:"keyid"`
+		KeyVal              keyval   `json:"keyval"`
+		Scheme              string   `json:"scheme"`
+		KeyIDHashAlgorithms []string `json:"keyid_hash_algorithms"`
+	}{}
```

Comment:
nit: pull these out into top level types - I don't think we gain much by defining these in the func themselves.

Reply:
Is this the same as tuf.Key below?

Reply:
> Is this the same as tuf.Key below?

Close but this supports `private`. I'm working through removing large chunks of this module. I've upstreamed the ed25519 signing to go-securesystemslib (and defined the key format there as well). Can we leave this in for now and I'll submit a separate patch for this specific package as a whole?

Reply:
I'd still pull it into it's own type + a comment to explain the difference, then we can clean it up once the other changes land.

Reply:
Done!